### PR TITLE
Add Javascript resource widget for dashboard

### PR DESCRIPTION
**Pull request by aabadie**
The idea is simple: embed a small javascript editor for nodes exposing a `js` resource: User can edit the script and send it to the corresponding node.

Added an example in the coap_test_node.py script.

### CODE REVIEW

**Comment by aabadie** [comment]
How to test locally:
1. Run the broker:
```
$ bin/aiot-broker
```
2. Run the dashboard web server
```
$ bin/aiot-dashboard
```
3. Run the coap gateway
```
$ bin/aiot-coap-gateway --coap-port=5684
```
4. Run the python test node
```
$ python utils/coap/coap-test-node.py --gateway-port=5684 --js
```

Play with the dashboard at http://localhost:8080

File: utils/coap/coap-test-node.py [hunk 1]
```diff
@@ -282,6 +284,44 @@ def render_put(self, request):
         return response
 
 
+class JavascriptResource(resource.Resource):
+    """Test node Javascript ressource."""
+
+    def __init__(self):
+        super(JavascriptResource, self).__init__()
+        self.script = """
+this.ledorange = saul.get_by_name("led");
```
Comment (bformet) [inline utils/coap/coap-test-node.py:293]: indentation looked weird to me at first, and then I noticed the triple quotes...

there is no perfect way to deal with multiline strings imo.
maybe [textwrap.dedent()](https://docs.python.org/3/library/textwrap.html#textwrap.dedent) do you know it ?

Comment (aabadie) [reply utils/coap/coap-test-node.py:293]: no I don't but thanks for pointing it. I think it's fine with the actual unindented triple quotes.

**Review by bformet — APPROVED** [review]
nice addition to the dashboard.
could be enhanced later with "click to enlarge" to make js code reading/editing easier.

**Comment by aabadie** [comment]
@bformet, can you "rebase & merge" ? (normally you have write access)

**Comment by bformet** [comment]
well apparently not

**Comment by aabadie** [comment]
Found a couple of remaining issues: directly pushed to master ;)

**Pull request merged**